### PR TITLE
Bug/import survey activity log order

### LINKED ISSUE #1424: SurveyImport: activity log
- Imported survey does not maintain same activity log 
test/e2e/tests/surveyImport.js

- infinite loading of activities in dashboard 
(see video: https://drive.google.com/file/d/10HAA7acewAJeevOF5zX-YErz0-z4t5_s/view?usp=sharing)

### PR DESCRIPTION
**Pull request by ramirobg94**
## Related Issue

Resolves #1424 

## Types of changes

- [ ] Docs change
- [ ] Refactoring
- [ ] Dependency upgrade
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Brief description of the changes proposed and/or how the issue(s) has(have) been solved.

The activity log should be imported in order to keep the ids

##### Do you consider this PR needs further testing?

- [x] No
- [ ] Yes

### CODE REVIEW

**Review by SteRiccio — DISMISSED** [review]

**Review by minotogna — APPROVED** [review]

File: server/modules/arenaImport/service/arenaImport/metaImportJobs/surveyCreatorJob.js [hunk 1]
```diff
@@ -45,8 +46,10 @@ export default class SurveyCreatorJob extends Job {
     const surveyId = Survey.getId(survey)
 
     const activities = await ArenaSurveyFileZip.getActivities(arenaSurveyFileZip)
-
-    await ActivityLogManager.insertManyBatch(this.user, surveyId, activities, this.tx)
+    const activitiesOrderById = activities.sort(
+      (activityA, activityB) => ActivityLog.getDateCreated(activityA) - ActivityLog.getDateCreated(activityB)
+    )
+    await ActivityLogManager.insertManyBatch(this.user, surveyId, activitiesOrderById, this.tx)
```
Comment (minotogna) [inline server/modules/arenaImport/service/arenaImport/metaImportJobs/surveyCreatorJob.js:52]: wouldn't it be better to export them in order already ?

Comment (ramirobg94) [reply server/modules/arenaImport/service/arenaImport/metaImportJobs/surveyCreatorJob.js:52]: Yep, Also we are reusing the same function that we use for the activitylog and in that case I think that system logs are not exported... so, yes I think that it is better add a new Repository function only to export the activity log

**Review by SteRiccio — DISMISSED** [review]

File: server/modules/activityLog/manager/activityLogManager.js [hunk 2]
```diff
@@ -74,10 +74,10 @@ const _getAvailableActivityTypes = async (surveyUuid, user) => {
   )(user)
 }
 
-export const fetch = async ({ user, surveyId, idGreaterThan, idLessThan, limit }) => {
+export const fetch = async ({ user, surveyId, idGreaterThan, idLessThan, limit, orderIdBy }) => {
```
Comment (ramirobg94) [inline server/modules/activityLog/manager/activityLogManager.js:77]:
```suggestion
export const fetch = async ({ user, surveyId, idGreaterThan, idLessThan, limit, orderBy }) => {
```

File: server/modules/activityLog/manager/activityLogManager.js [hunk 3]
```diff
@@ -74,10 +74,10 @@ const _getAvailableActivityTypes = async (surveyUuid, user) => {
...
   const surveyInfo = await SurveyRepository.fetchSurveyById(surveyId)
   const activityTypes = await _getAvailableActivityTypes(Survey.getUuid(surveyInfo), user)
-  return ActivityLogRepository.fetch({ surveyInfo, activityTypes, idGreaterThan, idLessThan, limit })
+  return ActivityLogRepository.fetch({ surveyInfo, activityTypes, idGreaterThan, idLessThan, limit, orderIdBy })
```
Comment (ramirobg94) [inline server/modules/activityLog/manager/activityLogManager.js:80]:
```suggestion
  return ActivityLogRepository.fetch({ surveyInfo, activityTypes, idGreaterThan, idLessThan, limit, orderBy })
```

File: server/modules/activityLog/repository/activityLogRepository.js [hunk 4]
```diff
@@ -70,6 +70,7 @@ export const fetch = async ({
   idGreaterThan = null,
   idLessThan = null,
   limit = 30,
+  orderIdBy = 'DESC',
```
Comment (ramirobg94) [inline server/modules/activityLog/repository/activityLogRepository.js:73]:
```suggestion
  orderBy = 'DESC',
```

File: server/modules/activityLog/repository/activityLogRepository.js [hunk 5]
```diff
@@ -250,7 +251,7 @@ export const fetch = async ({
     -- end of analysis activities part
 
     ORDER BY
-      l.id DESC`,
+      l.id ${orderIdBy}`,
```
Comment (ramirobg94) [inline server/modules/activityLog/repository/activityLogRepository.js:254]:
```suggestion
      l.id ${orderBy}`,
```

File: server/modules/survey/service/surveyService.js [hunk 6]
```diff
@@ -146,7 +146,7 @@ export const exportSurvey = async ({ surveyId, res, user }) => {
   const activityLogPathDir = FileUtils.join(prefix, 'activitylog')
   const activityLogPathFile = FileUtils.join(activityLogPathDir, 'activitylog.json')
 
-  const activityLog = await ActivityLogService.fetch({ user, surveyId, limit: 'ALL' })
+  const activityLog = await ActivityLogService.fetch({ user, surveyId, limit: 'ALL', orderIdBy: 'ASC' })
```
Comment (ramirobg94) [inline server/modules/survey/service/surveyService.js:149]:
```suggestion
  const activityLog = await ActivityLogService.fetch({ user, surveyId, limit: 'ALL', orderBy: 'ASC' })
```

**Review by minotogna — APPROVED** [review]

**Pull request merged**